src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenNullCheckedParameterTests.cs

```diff
@@ -1264,5 +1300,417 @@ .maxstack  1
+            }
+    }
+    static void Main()
+    {
+        Program p = new Program();
+        var c = Invoke(() => p.GetChars(string.Empty));
+        var e = Invoke(() => c.GetEnumerator());
+        Invoke(() => e.MoveNext());
+        c = Invoke(() => p.GetChars(null));
+    }
+    static T Invoke<T>(Func<T> f)
+    {
+        T t = default;
+        try
+        {
+            t = f();
+            Console.WriteLine(""ok"");
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine(e.GetType());
+        }
+        return t;
+    }
+}";
+            CompileAndVerify(source, expectedOutput: @"
+ok
+ok
+ok
+System.ArgumentNullException");
+        }
+
+        [Fact]
+        public void TestNullCheckedWithMissingType()
+        {
+            var source =
+@"
+class Program
+{
+    static void Main(string[] args!) { }
+}";
+            var comp = CreateCompilation(source);
+            comp.MakeMemberMissing(WellKnownMember.System_ArgumentNullException__ctorString);
+            comp.MakeTypeMissing(WellKnownType.System_ArgumentNullException);
+            comp.VerifyEmitDiagnostics(
+                    // (4,31): error CS0656: Missing compiler required member 'System.ArgumentNullException..ctor'
+                    //     static void Main(string[] args!) { }
+                    Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "args").WithArguments("System.ArgumentNullException", ".ctor").WithLocation(4, 31));
+        }
+
+        [Fact]
+        public void TestEmptyNullCheckedIterator1()
+        {
+            var source = @"
+using System.Collections.Generic;
+class C
+{
+    public static void Main() { }
+    static IEnumerable<char> GetChars(string s!)
+    {
+        yield break;
+    }
+}";
+            CompileAndVerify(source).VerifyIL("C.GetChars(string)", @"
+{
+    // Code size       22 (0x16)
+    .maxstack  1
+    IL_0000:  ldarg.0
+    IL_0001:  brtrue.s   IL_000e
+    IL_0003:  ldstr      ""s""
+    IL_0008:  newobj     ""System.ArgumentNullException..ctor(string)""
+    IL_000d:  throw
+    IL_000e:  ldc.i4.s   -2
+    IL_0010:  newobj     ""C.<GetChars>d__1..ctor(int)""
+    IL_0015:  ret
+}");
+        }
+
+        [Fact]
+        public void TestEmptyNullCheckedIterator2()
+        {
+            var source = @"
+using System.Collections.Generic;
+class C
+{
+    public static void Main() { }
+    static IEnumerator<char> GetChars(string s!)
+    {
+        yield break;
+    }
+}";
+            CompileAndVerify(source).VerifyIL("C.GetChars(string)", @"
+{
+    // Code size       21 (0x15)
+    .maxstack  1
+    IL_0000:  ldarg.0
+    IL_0001:  brtrue.s   IL_000e
+    IL_0003:  ldstr      ""s""
+    IL_0008:  newobj     ""System.ArgumentNullException..ctor(string)""
+    IL_000d:  throw
+    IL_000e:  ldc.i4.0
+    IL_000f:  newobj     ""C.<GetChars>d__1..ctor(int)""
+    IL_0014:  ret
+}");
+        }
+
+        [Fact]
+        public void TestNullCheckedOutString()
```

Comment:
>TestNullCheckedOutString [](start = 20, length = 24)

I don't think it's important that we have separate semantic and emit test classes, but since we do, it looks like all the tests from here down belong in the semantic test class.